### ISSUE TITLE
[Bug] Comment ending with backslash causes DL4006 fix to be ignored

### ISSUE THREAD

**Issue by tholok97** [hadolint/hadolint#294]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

```
FROM alpine:3.6

# \

# should solve DL4006
SHELL ["/bin/sh", "-o", "pipefail", "-c"]

# RUN with pipe. causes DL4006, but should be fixed by above SHELL
RUN echo "kaka" | sed 's/a/o/g' >> /root/afile
```

The above Dockerfile should pass hadolint.

### Actual behavior

The above Dockerfile still complains about DL4006. If I remove the `# \` it passes.

### Steps to reproduce the behavior

Put this in a Dockerfile:

```
FROM alpine:3.6

# The following comment makes hadolint still complain about DL4006
# \

# should solve DL4006
SHELL ["/bin/sh", "-o", "pipefail", "-c"]

# RUN with pipe. causes DL4006, but should be fixed by above SHELL
RUN echo "kaka" | sed 's/a/o/g' >> /root/afile

# The backslash does not seem to affect `docker build`, "koko" will be added to /root/afile here
# \
RUN echo "koko" >> /root/afile
```

Build and run with `sudo docker build -t test . && docker run test cat /root/afile`, the output will be:

```
Step 1/4 : FROM alpine:3.6
 ---> bb000f8b5177
Step 2/4 : SHELL ["/bin/sh", "-o", "pipefail", "-c"]
 ---> Using cache
 ---> 62a8b1a3c51d
Step 3/4 : RUN echo "kaka" | sed 's/a/o/g' >> /root/afile
 ---> Using cache
 ---> b672ab42ecc3
Step 4/4 : RUN echo "koko" >> /root/afile
 ---> Using cache
 ---> e586bd1b526f
Successfully built e586bd1b526f
Successfully tagged test:latest
koko
koko
```

The fact that "koko" is printed twice in my eyes shows that `docker build` does not ignore the next command after `# \` as it seems like Hadolint does.

Now run hadolint: `sudo docker run --rm -i hadolint/hadolint < Dockerfile`. The output will be `/dev/stdin:10 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it`.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.16.0-0-g475ad73
```

**Dockerfile** (if relevant)

```Dockerfile
FROM alpine:3.6

# The following comment makes hadolint still complain about DL4006
# \

# should solve DL4006
SHELL ["/bin/sh", "-o", "pipefail", "-c"]

# RUN with pipe. causes DL4006, but should be fixed by above SHELL
RUN echo "kaka" | sed 's/a/o/g' >> /root/afile

# The backslash does not seem to affect `docker build`, "koko" will be added to /root/afile here
# \
RUN echo "koko" >> /root/afile
```

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
Newline escape rules un Docker are arbitrary indeed.

What’s the need for finishing the line with a backslash?

**Comment by tholok97** [comment]
The issue popped up when I commented out a RUN command like this:
```
# RUN CATALINA_OPTS="\
# -Dwhatever=something \
# -Dotherthing=morestuff"

SHELL ["/bin/bash", "-o", "pipefail", "-c"]

# Install Filebeat
# TODO: Has to be configured!
RUN curl -sS -L -o filebeat.rpm https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-${FILEBEAT_VERSION}-x86_64.rpm && \
 echo "${FILEBEAT_SHA256}" filebeat.rpm | sha256sum -c -

```

Committed the change to see our pipeline output, but then the change never reach our tests because hadolint reported DL4006 in an earlier linting stage :/

So yes the issue can be avoided by keeping in mind not to put `\` at the end of comments, but felt it was good to report it anyways since ideally Hadolint should be able to handle it :)

(Really appreciate the tool btw!)

**Comment by lorenzo** [comment]
Thanks for your thoughtful ticket, I wanted to know if there was any reason I was missing for adding the backslash there.

Will look into fixing this soon.

**Comment by tholok97** [comment]
Awesome! Thank you for the quick reply.
